### PR TITLE
upload csiproxy binary to gcs with version information

### PR DESCRIPTION
**Pull request by jingxu97**
add logic to upload binary to gcs with version information

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,6 +11,12 @@ BUILD_DIR = build
 BUILD_TOOLS_DIR = $(BUILD_DIR)/tools
 GO_ENV_VARS = GO111MODULE=on GOOS=windows
 CSI_PROXY_API_GEN = $(BUILD_DIR)/csi-proxy-api-gen
+VERSION=$(shell git describe)
+REV=$(shell git describe --long --tags --match='v*' --dirty 2>/dev/null || git rev-list -n1 HEAD)
+
+$(info PULL_BASE_REF is $(PULL_BASE_REF))
```
Comment (msau42) [inline Makefile:17]: it doesn't look like're using these values in the bucket path. Is that right?

Comment (jingxu97) [reply Makefile:17]: actually I am still testing and verifying this part, so put WIP in this PR. sorry for the confusion.

**Comment by jingxu97** [comment]
@ddebroy @mauriciopoppe could you lgtm again? Thanks!

**Pull request merged**